pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -197,10 +196,15 @@ func (ku *Upgrader) upgradeAgentPools() error {
 			return err
 		}
 
-		var agentCount int
-		for _, app := range ku.ClusterTopology.DataModel.Properties.AgentPoolProfiles {
+		var agentCount, agentPoolIndex int
+		var agentOsType api.OSType
+		var agentPoolName string
+		for indx, app := range ku.ClusterTopology.DataModel.Properties.AgentPoolProfiles {
 			if app.Name == *agentPool.Name {
 				agentCount = app.Count
+				agentOsType = app.OSType
+				agentPoolName = app.Name
+				agentPoolIndex = indx
```

Comment:
Where is this being used?

Reply:
vmName, err := armhelpers.GetK8sVMName(agentOsType, ku.DataModel.Properties.HostedMasterProfile != nil,
				ku.NameSuffix, agentPoolName, agentPoolIndex, agentIndex)